File: BLPS_Lab1/src/main/java/org/example/blps_lab1/services/HotelService.java
package org.example.blps_lab1.services;

import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.example.blps_lab1.models.Hotel;
import org.example.blps_lab1.repositories.HotelRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@Slf4j
@AllArgsConstructor
public class HotelService {
    private final HotelRepository hotelRepository;

    public void createHotel(final Hotel hotel){
        hotelRepository.save(hotel);
        log.info("Created person: {}", hotel);
    }

    public void deleteHotel(final Long id){
        Optional<Hotel> deletingHotel = hotelRepository.findById(id);
        if(deletingHotel.isEmpty()){
            log.error("Hotel with id {} does not exist", id);
            throw new RuntimeException("Отель с таким id не существует");
        }
        hotelRepository.deleteById(id);
        log.info("Hotel deleted: {}", id);
    }

    public List<Hotel> getAllHotels(List<Hotel> hotel){
        var savedList = hotelRepository.saveAll(hotel);
        log.info("Saved {} hotels", savedList.size());
        return savedList;
    }
}

File: BLPS_Lab/src/main/kotlin/org/example/blps_lab/services/HotelService.kt
package org.example.blps_lab.services

import org.example.blps_lab.dto.HotelDto
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.example.blps_lab.models.Hotel
import org.example.blps_lab.repositories.HotelRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class HotelService(
    private val hotelRepository: HotelRepository
) {
    private val log: Logger = LoggerFactory.getLogger(HotelService::class.java)

    fun createHotel(hotel: Hotel) {
        hotelRepository.save(hotel);
        log.info("Hotel created: {}", hotel)
    }

    fun deleteHotel(hotelId: UUID) {
        val hotel = hotelRepository.findById(hotelId).orElseThrow(){
            log.warn("Hotel with id {} not found", hotelId)
            throw RuntimeException("Отель с таким id не существует")
        }
        hotelRepository.delete(hotel)
        log.info("Hotel with id {} deleted", hotelId)

    }

    fun getAllHotels(): List<Hotel>{
        val hotels = hotelRepository.findAll()
        log.info("All hotels retrieved: {}", hotels)
        return hotels
    }

    fun updateHotel(hotelId: UUID, hotelDto: HotelDto): Hotel{
        val hotelBefore = hotelRepository.findById(hotelId).orElseThrow(){
            log.warn("Hotel with id {} not found", hotelId)
            throw RuntimeException("Отель с таким id не существует")
        }

        hotelBefore.apply {
            hotelName = hotelDto.hotelName
            price = hotelDto.price
            dayOFArrival = hotelDto.dayOfArrival
            dayOfDeparture = hotelDto.dayOfDeparture
            guestsCount = hotelDto.guestCount
            address = hotelDto.address
        }

        val updatedHotel = hotelRepository.save(hotelBefore)
        log.info("Hotel with id {} updated", hotelId)
        return updatedHotel
    }
}